Maker-Agents/Governance-Module/src/hat.fetcher.spec.ts

```diff
@@ -9,8 +9,16 @@ describe("HatFetcher test suite", () => {
   const addr2: string = createAddress("0x2");
   const addr3: string = createAddress("0x3");
   const { setHat, setApproval, mockProvider } = mockWrapper(chief);
-  let hatFetcher: HatFetcher = new HatFetcher(chief, mockProvider as any);
+  let mockAddressFetcher: any;
+  let hatFetcher: HatFetcher;
 
+  beforeAll(() => {
+    mockAddressFetcher = {
+      getChiefAddress: jest.fn(),
+      chiefAddress: chief,
+    };
+    hatFetcher = new HatFetcher(mockAddressFetcher, mockProvider as any);
+  });
   beforeEach(() => mockProvider.clear());
 
   it("should fetch the hat properly", async () => {
```

Comment:
Should there be a `mockProvider.clear();` before L42? Similar to the one found in L30. (GH wouldn't let me write a comment on those lines.)